app/services/idv/steps/threat_metrix_step_helper.rb

```diff
@@ -46,6 +46,22 @@ def threatmetrix_iframe_url(session_id)
           session_id: session_id,
         )
       end
+
+      def log_irs_tmx_fraud_check_event(result)
+        return unless FeatureManagement.proofing_device_profiling_collecting_enabled?
+        success = result[:review_status] == 'pass'
+
+        if !success && result[:response_body] && result[:response_body][:tmx_summary_reason_code]
+          failure_reason = {
+            tmx_summary_reason_code: result[:response_body][:tmx_summary_reason_code],
+          }
+        end
```

Comment:
An option to simplify
```suggestion
        if !success && (tmx_summary_reason_code = result.dig(:response_body, :tmx_summary_reason_code))
          failure_reason = {
            tmx_summary_reason_code: tmx_summary_reason_code,
          }
        end
```